src/shared/ThemeContext/ThemeContext.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { createContext, FC, ReactNode, useEffect, useState } from 'react'
+
+export enum Theme {
+  LIGHT = 'light',
+  DARK = 'dark',
+}
+
+interface ThemeContextProps {
+  theme: Theme
+  setTheme: (theme: Theme) => void
+}
+
+const defaultThemeContext: ThemeContextProps = {
+  theme: Theme.LIGHT,
+  setTheme: () => {},
+}
+
+export const ThemeContext =
+  createContext<ThemeContextProps>(defaultThemeContext)
+
+interface ThemeContextProviderProps {
+  children: ReactNode
+}
+
+export const ThemeContextProvider: FC<ThemeContextProviderProps> = ({
+  children,
+}) => {
+  const currentTheme = (localStorage.getItem('theme') as Theme) || Theme.LIGHT
+  const [theme, setTheme] = useState<Theme>(currentTheme)
+
+  useEffect(() => {
+    document.body.classList.remove(Theme.LIGHT, Theme.DARK)
+    document.body.classList.add(theme)
```

Comment:
Can we add some checks in here to make sure `document.body` exists, just to make sure there's wacky potential errors, that could pop up

Reply:
added!